src/components/dialogs/generator-creation-dialog.js

```diff
@@ -219,6 +251,72 @@ const GeneratorCreationDialog = ({
         }
     };
 
+    const handleSelectionChange = (element) => {
+        let msg;
+        fetchGeneratorInfos(studyUuid, selectedNodeUuid, element.id).then(
+            (response) => {
+                if (response.status === 200) {
+                    response.json().then((generator) => {
+                        setFormValues(null);
```

Comment:
Is it usefull ? formValues is set just after ...

Reply:
Yes it's needed. Some useEffects trigger on value change, if we copy in a row 2 equipment that share for exemple the same voltageLevel it won't reset the busbarsection because the voltageLevel didn't change hence we need to first reset the fields.